tests/testthat/test-deterministic.R

```diff
@@ -362,3 +362,37 @@ test_that("Can offset the initial likelihood", {
   expect_identical(p1$history(), p2$history())
   expect_identical(p1$state(), p2$state())
 })
+
+
+test_that("Can use compiled compare", {
+  dat <- example_sir()
+  p1 <- particle_deterministic$new(dat$data, dat$model, dat$compare, dat$index)
+  p2 <- particle_deterministic$new(dat$data, dat$model, NULL, dat$index)
+  pars <- list(exp_noise = Inf)
+  ## there's a discrepency here which ironically goes away with
+  ## non-infinite noise due to combination of sum over small numbers
+  ## and slightly different density calculation accuracy.
+  expect_equal(p1$run(pars), p2$run(pars), tolerance = 2e-5)
```

Comment:
Different example/non-Inf noise simpler then?

Reply:
Using an Inf noise keeps the calculation deterministic, even if not quite at the accuracy we'd like.  Getting the details out as to why this is slightly off would be potentially interesting but fairly sure that it's just that we have some poor accuracy in the densities when very unlikely (the R versions of these have lots of support to make these accurate but we can't use them due to thread safety)